mautic/mautic
# Display list of contacts on click in Click stats table across all channel

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Yes
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/1883
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR added clickable number of unique clicks across all channels (email, sms, notification, dynamicContent) Click stats table. Number of unique clicks will redirect to contact list with  related filter. Same like search commands: email_sent, email_read added few months ago.  This PR added new search commands  page_id, page_source, page_source_id
Also this PR fixed issue if somebody wrote same search command twice and more time. This lines fixed it https://github.com/mautic/mautic/pull/5849/files#diff-3fa38c644457bfa255aabbc8bd7e5291R1090

Preview in email details template: 

![image](https://user-images.githubusercontent.com/462477/37739302-ff77b918-2d59-11e8-91eb-863e680b5d66.png)


[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1.  Try search with two same commands
![image](https://user-images.githubusercontent.com/462477/37739479-815537e4-2d5a-11e8-8090-52dc0c48e619.png)
2. If you use dev mode, error return. Or check logs for error something like 

> The given alias xx is not unique in FROM 

#### Steps to test this PR:
1. Apply PR and repeat steps
2. See no error 

#### Steps to test this feature request:

1. Apply PR and create email, notification, sms, dynamicContent with some click statistics
2. Go to all created entity to details
3. Scroll to click counts table and click on any number of unique clicks
4. Click should redirect to contacts page with search commands and related results
5. Check if all results works properly
